notebook/static/notebook/js/cell.js

```diff
@@ -515,14 +516,28 @@ define([
     };
 
     /**
-     * is the cell deletable? only false (undeletable) if
-     * metadata.deletable is explicitly false -- everything else
+     * is the cell edtitable? only false (readonly) if
+     * metadata.editable is explicitly false -- everything else
      * counts as true
      *
+     * @method is_editable
+     **/
+    Cell.prototype.is_editable = function () {
+        if (this.metadata.editable === false) {
+            return false;
+        }
+        return true;
+    };
+
+    /**
+     * is the cell deletable? only false (undeletable) if
+     * metadata.deletable is explicitly false or if the cell is not
+     * editable -- everything else counts as true
+     *
      * @method is_deletable
      **/
     Cell.prototype.is_deletable = function () {
-        if (this.metadata.deletable === false) {
+        if (this.metadata.deletable === false || !this.is_editable()) {
```

Comment:
> I'm fine with changing the default but if we do that (go from deletable to undeletable) could we build in something that would automatically update the notebook metadata if the old deletable metadata is found?

@jhamrick This block will handle cases where `metadata.deletable` is undefined (`is_deletable()` will return `true`). Do you think that we need to write that value back to the metadata? It seems like this would only be necessary if a notebook were to be used in another client that doesn't define an `is_deletable()` equivalent.